File: Iniciante/PHP/exercise_1040.php
<?php

function floorDec($val, $precision = 2) {
    if ($precision < 0) {
        $precision = 0;
    }
    $numPointPosition = intval(strpos($val, '.'));
    if ($numPointPosition === 0) {
        //$val is an integer
        return $val;
    }
    return floatval(substr($val, 0, $numPointPosition + $precision + 1));
}



$valores = readline("Insira as 4 notas: ");

list($nota1, $nota2, $nota3, $nota4) = sscanf($valores, "%f %f %f %f");

$media = ($nota1*2+$nota2*3+$nota3*4+$nota4*1)/10;

print_r("Media: ".number_format(floorDec($media, 1), 1)."\n");

switch ($media) {
    case ($media >= 7):
        print_r("Aluno aprovado.\n");
        break;
    case ($media < 5):
        print_r("Aluno reprovado.\n");
        break;
    case ($media >= 5 && $media < 7):
        print_r("Aluno em exame.\n");

        $notaExame = floatval(fgets(STDIN));

        print_r("Nota do exame: ".number_format($notaExame, 1)."\n");

        $novaMedia = ($media+$notaExame)/2;

        if ($novaMedia >= 5) {
            print_r("Aluno aprovado.\n");
        } else {
            print_r("Aluno reprovado.\n");
        }

        print_r("Media final: ".number_format($novaMedia, 1)."\n");
        break;
}

//Concluído

File: Iniciante/Python/exercises from 1000 to 1099/exercise_1040.py
nota1, nota2, nota3, nota4 = [float("{:.1f}".format(float(x))) for x in input().split()]

media = (nota1*2+nota2*3+nota3*4+nota4*1)/10

print(''.join(['Media: ', "{:.1f}".format(media)]))

if media >= 7:
    print('Aluno aprovado.')
elif media < 5:
    print('Aluno reprovado.')
elif media >= 5 and media < 7:
    print('Aluno em exame.')

    notaExame = float("{:.1f}".format(float(input())))
    print(''.join(['Nota do exame: ', "{:.1f}".format(notaExame)]))

    media = (media+notaExame)/2

    if media >= 5:
        print('Aluno aprovado.')
    else:
        print('Aluno reprovado.')

    print(''.join(['Media final: ', "{:.1f}".format(media)]))